docker/README.md

```diff
@@ -0,0 +1,25 @@
+# Maintaining PBS Pro Dockerfiles
+
+## Prerequisite
+You need to know how to write Dockerfiles using multi-stage builds. If not please at least read the following materials before proceeding:
+1. [Dockerfile reference](https://docs.docker.com/engine/reference/builder/)
+2. [Best practices for writing Dockerfile](https://docs.docker.com/engine/userguide/eng-image/dockerfile_best-practices/)
+3. [Use multi-stage builds](https://docs.docker.com/engine/reference/builder/)
+
+## Directory Structure
+Each directory corresponds to a supported linux distribution. For example, Dockerfiles for building an image based on CentOS version 7 are under the `centos7` directory. Each directory contains at least three files, a `Dockerfile.base`, a `Dockerfile.build` and `build.sh`. Please adhere to this structure when you create Dockerfiles for a new distribution.
+
+### `Dockerfile.base`
+This file defines an image that will be used as builder in the docker build process. It must be based on an official linux image and have the necessary packages for building PBS Pro installed. It should copy `build.sh` under the same directory to `/` and have an `ONBUILD RUN bash /build.sh` command so that a PBS Pro build is triggered when this image is used as a base image.
```

Comment:
This file defines the base image that will be used to build PBS Pro.
It should copy the corresponding 'build.sh' to '/' and contain the 'ONBUILD RUN bash /build.sh' directive to trigger the PBS Pro build.